pages/contests/[id].tsx

```diff
@@ -0,0 +1,137 @@
+
+  const handleImageClick = (image) => {
+    setSelectedImage(image)
+  }
+
+  const closeImageModal = () => {
+    setSelectedImage(null)
+  }
+
+  useEffect(() => {
+    const handleClickOutside = (event) => {
+      if (event.target.classList.contains('modal-overlay')) {
+        closeImageModal()
+      }
+    }
+
+    document.addEventListener('click', handleClickOutside)
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside)
+    }
+  }, [])
+
+  return (
+    <>
+      {loadingDetail && <p>Loading</p>}
+      {errorDetail && <p>Error while retrieving the contest</p>}
+      {!loadingDetail && !errorDetail && (
+        <>
+          <div className='w-full  justify-center h-full bg-gray-200'>
+            <div className='bg-white p-8 rounded-lg shadow-lg'>
+              <div className='flex justify-center items-center'>
+                <img
+                  src={contestDetail?.cover_picture?.picture_path}
+                  alt='Imagem'
+                  className='object-fill h-64 w-96'
+                />
+              </div>
+              <p className='text-center mt-4 text-2xl font-bold'>
+                {contestDetail?.title}
+              </p>
+              <p className='text-center mt-2 text-lg'>
+                {contestDetail?.description}
+              </p>
+              <div className='mt-6'>
+                <div className='mt-2 flex flex-wrap'>
+                  {loadingSubmission && <p>Loading</p>}
+                  {errorSubmission && (
+                    <p>Error while retrieving the contest submissions</p>
+                  )}
+
+                  {!loadingSubmission && !errorSubmission && (
+                    <>
+                      {submissionList?.map((image) => (
+                        <div
+                          key={image.id}
+                          className='w-1/4 mt-2 flex flex-col items-center mx-2'
+                        >
+                          <img
+                            src={image.picture.picture_path}
+                            alt={`Image ${image.id}`}
+                            className='w-full h-auto max-h-60'
+                            onClick={() => handleImageClick(image)}
+                          />
+                          <p className='mt-2 text-center'>
+                            User: {image.picture.user.name_first}{' '}
+                            {image.picture.user.name_last}
+                          </p>
+                          <div className=' flex items-center justify-center'>
+                            <button
+                              className='mt-2 text-gray-700 bg-gray-500 text-white px-3 py-2 rounded-2xl font-medium cursor-pointer mr-2'
+                              type='submit'
+                            >
+                              Vote
+                            </button>
+                          </div>
+                        </div>
```

Comment:
this should be a separate component

Reply:
a quick way to see if something should be a component is if it's used inside a `.map()` or any sort of iteration